include/sampling/random_point_generators.hpp

```diff
@@ -187,10 +187,10 @@ struct LogconcaveRandomPointGenerator
                       RandomNumberGenerator &rng,
                       NegativeGradientFunctor &F,
                       NegativeLogprobFunctor &f,
-                      Parameters &parameters)
+                      Parameters &parameters,
+                      Walk &walk)
```

Comment:
what is the difference here wrt to other walks? 
@TolisChal

Reply:
The structure `walk` always is given as input. The set of parameters also is different (function, gradient oracles etc).
I dont know if my answer is complete.

Reply:
@TolisChal is right. I wanted to keep the same `walk` after the burn-in phase (i.e. the above function is called two times; one for burn-in, then the points are cleared, and one for the actual walk). We also need the `walk` since we have computed the adaptive step size.